### PR TITLE
feat: Update README

### PR DESCRIPTION
**Pull request by SahilK-027**
## Clickup
[clickup link](https://app.clickup.com/t/86cx5ak7e)

## Summary by CodeRabbit

- **Documentation**
	- Restructured and enhanced the `README.md` file for better clarity and organization.
	- Updated the title to "Fyle Mobile App" and added an introductory paragraph outlining the repository's purpose.
	- Introduced a "Table of Contents" for easier navigation with links to various sections.
	- Expanded sections on prerequisites, quick setup, environment setup, troubleshooting, project structure, testing, and running on Android/iOS devices.
	- Added detailed instructions for push notifications and a "Further Help" section for user support.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,118 +1,342 @@
...
+
+<br/>
+
+## ❓ Troubleshooting
+
+Here are some common issues and how to fix them:
+
+1. If you encounter the any similar error like this:
+   `Property 'LIVE_UPDATE_APP_VERSION' does not exist on type ...`
+
```
[the remark this answers is not in the document]
Comment (SahilK-027) [reply README.md:89]: Thanks I will fix it!

**Review by Chethan-Fyle — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,118 +1,342 @@
...
+- [📱 Running on Devices](#-running-on-devices)
+  - [Android](#android)
+  - [iOS](#ios)
+- [📬 Push Notifications](#-push-notifications)
+  - [iOS Setup](#ios-setup)
+- [📊 Deployment](#-deployment)
+  - [Running Appflow workflow manually on a private branch](#running-appflow-workflow-manually-on-a-private-branch)
+- [🙏 Further Help](#-further-help)
+
+<br/>
```
[the remark this answers is not in the document]
Comment (SahilK-027) [reply README.md:23]: Using <br/> in other repos as well

File: README.md [hunk 3]
```diff
@@ -2,117 +1,342 @@
...
-Please install node v14.17.2 or above via nvm.
+## 📑 Table of Contents
 
-## How to run this locally?
+- [🔧 Prerequisites](#-prerequisites)
+- [✨ Quick Setup](#-quick-setup)
+- [👀 Environment Setup](#-environment-setup)
+- [❓ Troubleshooting](#-troubleshooting)
+- [📂 Project Structure](#-project-structure)
+- [🧪 Testing](#-testing)
```
[the remark this answers is not in the document]
Comment (SahilK-027) [reply README.md:12]: `[🧪 Testing](#️-testing)` How will adding extra space after # will fix it?

File: README.md [hunk 4]
```diff
@@ -1,118 +1,343 @@
...
+}
+```
+
+4. Enable Push Notifications in Xcode:
+
+- Open your project in Xcode.
+- Go to the "Signing & Capabilities" tab, and ensure that Push Notifications are enabled.
+- If you encounter any issues, refer to the following articles for troubleshooting:
+  - [The Push Notifications Guide for Ionic & Capacitor](https://devdactic.com/push-notifications-ionic-capacitor/)
+  - [Using Push Notifications with Firebase in an Ionic + Angular App](https://capacitorjs.com/docs/guides/push-notifications-firebase)
```
Comment (arjunaj5) [inline README.md:319]: we've removed push notifications, can remove this section

**Review by arjunaj5 — APPROVED** [review]

**Pull request merged**